internal/mode/static/manager.go

```diff
@@ -73,16 +80,118 @@ func StartManager(cfg config.Config) error {
 		return fmt.Errorf("cannot build runtime manager: %w", err)
 	}
 
+	hc := &healthChecker{}
+	if cfg.HealthConfig.Enabled {
+		if err := mgr.AddReadyzCheck("readyz", hc.readyCheck); err != nil {
+			return fmt.Errorf("error adding ready check: %w", err)
+		}
+	}
+
 	recorderName := fmt.Sprintf("nginx-kubernetes-gateway-%s", cfg.GatewayClassName)
 	recorder := mgr.GetEventRecorderFor(recorderName)
 	logLevelSetter := newZapLogLevelSetter(cfg.AtomicLevel)
 
+	ctx := ctlr.SetupSignalHandler()
+
+	controlConfigNSName := types.NamespacedName{
+		Namespace: cfg.Namespace,
+		Name:      cfg.ConfigName,
+	}
+	if err := registerControllers(ctx, cfg, mgr, recorder, logLevelSetter, eventCh, controlConfigNSName); err != nil {
+		return err
+	}
+
+	// protectedPorts is the map of ports that may not be configured by a listener, and the name of what it is used for
+	protectedPorts := map[int32]string{
```

Comment:
I think we need to add the `HealthConfig.Port` to this map. It's used in a later check to ensure no Listeners are configured using ports already in use by the Pod.